lib/LambdaHelper.js

```diff
@@ -118,7 +123,7 @@ var _createProxy = function (executionStore, queueInitializedPromise, promise) {
 
         });
 
-        promise.catch(function () {
+        uploadPromise.catch(function () {
             throw "Could not upload function to S3";
```

Comment:
I know it's not part of the PR but this is crap. The throw is not part of the synchronous workflow